### PR TITLE
Remove dead ends and return DiscreteTrajectory from Equipotential

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: physics/equipotential_body.hpp [hunk 1]
```diff
@@ -73,11 +73,18 @@ auto Equipotential<InertialFrame, Frame>::ComputeLine(
...
+        auto const& [double_q, double_β] = state.y;
+        DependentVariables values;
+        auto& [q, β] = values;
+        q = double_q.value;
+        β = double_β.value;
+        DependentVariableDerivatives derivatives;
+        RightHandSide(
+            binormal, position, t, state.s.value, values, derivatives);
+        auto const& [qʹ, βʹ] = derivatives;
+        equipotential.Append(Instant() + state.s.value * (1 * Second),
```
Comment (pleroy) [inline physics/equipotential_body.hpp:86]: Let's have a constant for this `1 * Second`, if only to make it easier to locate the places where we have a cheezy conversion to times/speeds.  (Also on the next line.)

File: physics/equipotential_test.cpp [hunk 2]
```diff
@@ -540,11 +488,8 @@ TEST_F(EquipotentialTest, RotatingPulsating_GlobalOptimization) {
           energy);
       for (auto const& line : lines) {
         all_positions.back().back().emplace_back();
-        all_βs.back().back().emplace_back();
-        for (auto const& state : line) {
-          auto const& [position, β] = state;
-          all_positions.back().back().back().push_back(position);
-          all_βs.back().back().back().push_back(β);
+        for (auto const& [s, dof] : line) {
+          all_positions.back().back().back().push_back(dof.position());
```
Comment (pleroy) [inline physics/equipotential_test.cpp:492]: Could we give names to all these backs?

File: physics/equipotential_test.cpp [hunk 3]
```diff
@@ -560,11 +505,8 @@ TEST_F(EquipotentialTest, RotatingPulsating_GlobalOptimization) {
         approx_l2_energy);
     for (auto const& line : lines) {
       all_positions.back().back().emplace_back();
-      all_βs.back().back().emplace_back();
-      for (auto const& state : line) {
-        auto const& [position, β] = state;
-        all_positions.back().back().back().push_back(position);
-        all_βs.back().back().back().push_back(β);
+      for (auto const& [s, dof] : line) {
+        all_positions.back().back().back().push_back(dof.position());
```
Comment (pleroy) [inline physics/equipotential_test.cpp:509]: Same.

File: physics/equipotential_test.cpp (hunk 2 above)
Comment (eggrobin) [reply physics/equipotential_test.cpp:492]: 🔙🔙🔙🔙

**Comment by eggrobin** [comment]
retest this please

**Pull request merged**